flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/State.java

```diff
@@ -38,10 +40,16 @@
 	private final String name;
 	private final StateType stateType;
 	private final Collection<StateTransition<T>> stateTransitions;
+	private final MatchingBehaviour matchingBehaviour;
```

Comment:
The `State` shouldn't have to know the `MatchingBehaviour`. They should be agnostic to the matching behaviour.